src/server/qgsconfigcache.h

```diff
@@ -30,6 +31,44 @@
 #include "qgsproject.h"
 #include "qgsserversettings.h"
 
+#ifndef SIP_RUN
+
+class QgsConfigCache;
+
+/**
+ * \ingroup server
+ * \brief Abstract base class for implementing
+ * cache invalidation strategy
+ *
+ * \since QGIS 3.24
+ */
+class SERVER_EXPORT QgsAbstractCacheStrategy
+{
+  public:
+    //! The name of the strategy
+    virtual QString name() const = 0;
+
+    /**
+     * Called when an entry is removed from cache.
+     * \param path The path of the project
+     */
+    virtual void entryRemoved( const QString &path ) = 0;
+
+    /**
+     * Called when an entry is removed from cache.
+     * \param path The path of the project
+     */
+    virtual void entryInserted( const QString &path ) = 0;
+
+    //! Attache cache to this strategy
+    virtual void attach( QgsConfigCache &cache ) = 0;
```

Comment:
> I have to disagree with @pblottiere on this point, QObject should be passed as pointers, not reference. That's how (almost) all QGIS handles this. And Qt as well.

On the Qt side I don't know but it's true on QGIS side. However I never liked it ^^. I think it's more robust to use reference instead of pointer because we cannot delete it (even by accident).

So what is the technical reason behind this choice?

Reply:
> Could we just drop this method attach and pass this through the constructor?

Not really, or the initialisation from settings has to be done in the `QgsConfig` cache constructor.

So pointer or not pointer ?

Reply:
> So pointer or not pointer ?

I think we can blindly trust @m-kuhn of course (no pointer). But I'm just curious to understand why :).

Reply:
> of course (no pointer) 

Hu, you mean I should revert this to pointer instead reference ?

Reply:
> Hu, you mean I should revert this to pointer instead reference ?

Yep sorry, it was unclear... -> revert reference and use pointer instead (sorry for the inconvenience)

Reply:
Reverted to pointer

Reply:
> > I have to disagree with @pblottiere on this point, QObject should be passed as pointers, not reference. That's how (almost) all QGIS handles this. And Qt as well.
> 
> On the Qt side I don't know but it's true on QGIS side. However I never liked it ^^. I think it's more robust to use reference instead of pointer because we cannot delete it (even by accident).
> 
> So what is the technical reason behind this choice?

I can only offer "soft" reasons

- When a `QObject` is created it's often used with a `parent` in which case it will be created with `new` and stored in a pointer. So its members are called with `->` so it's strange to change the pattern to a `.` after passing it to a method
- Pointers can also point to `nullptr`
- Qt constructs like `connect` expect pointers
- You shouldn't delete things you don't have ownership of by accident anyway ;-)
- When used as return values, with pointers there is no risk to accidentally take a copy
- Not sure if `QList<QObject &>` works actually?
- Basically I connect the concept of "identity" (you don't copy) to pointer and "value" (you copy whenever you want) to ref/variable (see https://doc.qt.io/qt-5/object.html)

Reply:
@m-kuhn ok thanks for the explanation :+1: